### PR TITLE
Handle empty strings in serializer

### PR DESCRIPTION
**Pull request by joakim-hove**
See discussion here: https://github.com/OPM/opm-common/pull/1895#pullrequestreview-479985190

### CODE REVIEW

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
@bska : opionions on this?

**Comment by bska** [comment]
> opinions on this?

I've looked at it and tested it locally.  It definitely fixes the test failure I got in my checked build and I don't see any obviously missing pieces either.  I'm going to merge this now.

File: opm/common/utility/Serializer.hpp [hunk 1]
```diff
@@ -86,20 +91,26 @@ class Serializer {
...
 template<>
 std::string inline Serializer::get<std::string>() {
     std::string::size_type length = this->get<std::string::size_type>();
-    this->pos += length;
-    return {std::addressof(this->buffer[this->pos - length]), length};
+    if (length == 0)
+        return std::string{};
+
+    this->read_pos += length;
+    return {std::addressof(this->buffer[this->read_pos - length]), length};
```
Comment (blattms) [inline opm/common/utility/Serializer.hpp:113]: Mmh, are we sure that this works in the general case? This calls the constructor with the `const char*` parameter and hence  a C string. IMHO a c string should be `\0` delimited, but it seems like we never write the delimiter. Wouldn't we need to use:
```
this->pack(value.c_str(), value.size()+1);
```
above to also write that zero?

Comment (blattms) [reply opm/common/utility/Serializer.hpp:113]: Oversaw the length argument.

**Pull request merged**